src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.kaziamyr.onlinebookstore.repository.impl;
+
+import com.kaziamyr.onlinebookstore.model.Book;
+import com.kaziamyr.onlinebookstore.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
```

Comment:
You have Spring data dependency, I think you could use `@Transactional` and simply use `Session` with try-with-resources without manual transaction handling.